### DISCUSSION TITLE
Is there any way to turn off the  Extented Query Protocol?

### DISCUSSION THREAD

**Discussion by sakurai-ryo** [psycopg/psycopg#636] [Q&A] [answered]
Hi team, Thank you for providing this great library.

My team are currently using psycopg3 to connect from AWS Lambda to RDS Proxy.
However, when using the Extented Query Protocol, a phenomenon called "pinning" occurs and the connection pool stops working.
https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/rds-proxy-managing.html#rds-proxy-pinning:~:text=Conditions%20that%20cause%20pinning%20for%20RDS%20for%20PostgreSQL

I would like to know if there is any way to turn off the Extented Query Protocol.
Also, if the Extented Query Protocol cannot be turned off, would psycopg2 be the best way to do this?

**Comment by dvarrazzo** [comment, marked as answer]
Hello,

By the look of it, it's not the extended query protocol to get in the way,
but prepared statements, because the problem with proxies is coordinating
the state between server sessions and client. All the example of
problematic statements (set, advisory locks, sequence functions) hint at
that.

Can you try turning prepared statements off by using
`prepare_threshold=None` as connect() params?

https://www.psycopg.org/psycopg3/docs/advanced/prepare.html#prepared-statements

Another possibility is to use client-side binding cursors, which, as side
effect don't use the extended protocol and altogether behave similarly to
psycopg2, but it's a more brutal solution.

https://www.psycopg.org/psycopg3/docs/advanced/cursors.html#client-side-binding-cursors

**Reply by sakurai-ryo** [reply]
Thank you so much for letting me know.
I will try it.
